taskflow/algorithm/for_each.hpp

```diff
@@ -36,16 +38,18 @@ TF_FORCE_INLINE auto make_for_each_task(B beg, E end, C c, P&& part = P()) {
       size_t chunk_size;
       for(size_t w=0, curr_b=0; w<W && curr_b < N; ++w, curr_b += chunk_size) {
         chunk_size = part.adjusted_chunk_size(N, W, w);
-        launch_loop(W, w, rt, [=, &c, &part] () mutable {
-          part.loop(N, W, curr_b, chunk_size,
-            [&, prev_e=size_t{0}](size_t curr_b, size_t curr_e) mutable {
-              std::advance(beg, curr_b - prev_e);
-              for(size_t x = curr_b; x<curr_e; x++) {
-                c(*beg++);
+        launch_loop(W, w, rt, [=, &c, &part,&task_wrapper] () mutable {
+          task_wrapper([&]{
+            part.loop(N, W, curr_b, chunk_size,
+              [&, prev_e=size_t{0}](size_t curr_b, size_t curr_e) mutable {
+                std::advance(beg, curr_b - prev_e);
+                for(size_t x = curr_b; x<curr_e; x++) {
+                  c(*beg++);
```

Comment:
Shouldn't we execute wrapper for this thread?

Reply:
I mean for general case, nobody knows if user-supplied wrapper can be skipped for parent thread.